### ISSUE TITLE
AFHMM & AFHMM+SAC - Training chunkwise only uses the last chunk

### ISSUE THREAD

**Issue by levaphenyl** [nilmtk/nilmtk-contrib#42]
Each call of `partial_fit` for AFHMM and AFHMM+SAC is overriding the results of the previous calls. Thus, a chunkwise training of those algorithms does not work as intended. The current implementation trains only on the last chunk.

The model parameters for AFHMM and AFHMM+SAC are `self.means_vector`, `self.pi_s_vector` and `self.transmat_vector`. However, these attributes are assigned [again](https://github.com/nilmtk/nilmtk-contrib/blob/master/nilmtk_contrib/disaggregate/afhmm.py#L90) and [again](https://github.com/nilmtk/nilmtk-contrib/blob/master/nilmtk_contrib/disaggregate/afhmm_sac.py#L115) without considering their previous values at each call of `partial_fit`.

**Comment by Rithwikksvr** [comment]
Yes @levaphenyl , you are right. We didn't design this to support chunk wise training. So, I think thats the reason

**Comment by Rithwikksvr** [comment]
The reason being the algorithm is more time consuming on the test set, and is not very intensive in the training
